python_modules/dagster/dagster_tests/core_tests/structured_config_tests/test_structured_config_types.py

```diff
@@ -439,3 +448,79 @@ def a_job():
         }
     )
     assert executed["yes"]
+
+
+def test_struct_config_optional_map() -> None:
+    class AnOpConfig(Config):
+        an_optional_dict: Optional[Dict[str, int]]
+
+    executed = {}
+
+    @op
+    def a_struct_config_op(config: AnOpConfig):
+        executed["yes"] = True
+
+    assert print_config_type_to_string(
+        a_struct_config_op.config_schema.config_type
+    ) == print_config_type_to_string(
+        Shape(fields={"an_optional_dict": DagsterField(Noneable(Map(str, IntSource)))})
+    )
+
+    @job
+    def a_job():
+        a_struct_config_op()
+
+    a_job.execute_in_process(
+        {"ops": {"a_struct_config_op": {"config": {"an_optional_dict": {"foo": 1, "bar": 2}}}}}
+    )
+    assert executed["yes"]
+
+    a_job.execute_in_process()
+    assert executed["yes"]
+
+    a_job.execute_in_process(
+        {"ops": {"a_struct_config_op": {"config": {"an_optional_dict": None}}}}
+    )
+    assert executed["yes"]
```

Comment:
don't you need to reset `executed` between executions in order to get this to work

Reply:
yes indeed